cpp/src/io/parquet/parquet.hpp

```diff
@@ -495,6 +646,53 @@ ParquetFieldStructFunctor<T> ParquetFieldStruct(int f, T &v)
   return ParquetFieldStructFunctor<T>(f, v);
 }
 
+/**
+ * @brief Functor to read a union member from CompactProtocolReader
+ *
+ * @return True if field types mismatch or if the process of reading a
+ * union member fails
+ */
+template <typename T, bool is_empty = false>
+class ParquetFieldUnionFunctor {
+  int field_val;
+  bool &is_set;
+  T &val;
+
+ public:
+  ParquetFieldUnionFunctor(int f, bool &b, T &v) : field_val(f), is_set(b), val(v) {}
+
+  inline bool operator()(CompactProtocolReader *cpr, int field_type)
+  {
+    return (field_type != ST_FLD_STRUCT || !(is_set = true, cpr->read(&val)));
+  }
+
+  int field() { return field_val; }
+};
+
+template <typename T>
+struct ParquetFieldUnionFunctor<T, true> {
+  int field_val;
+  bool &is_set;
+  T &val;
+
+ public:
+  ParquetFieldUnionFunctor(int f, bool &b, T &v) : field_val(f), is_set(b), val(v) {}
+
+  inline bool operator()(CompactProtocolReader *cpr, int field_type)
+  {
+    return (field_type != ST_FLD_STRUCT ||
+            !(is_set = true, cpr->skip_struct_field(field_type), true));
```

Comment:
I think this what the code is trying to do here :
```
if (field_type != ST_FLD_STRUCT) {
  return true;
} else {
  is_set = true;
  cpr->skip_struct_field(field_type);
  return false;
}
```
Is the comma operator used like this elsewhere in cudf to make lines compact? If not, other people might find it hard to decipher what is going on at a quick glance even though it looks quite elegant.

Reply:
Right. Will update to if, else.